### PR TITLE
validate the contents of seed before creating a Wallet object

### PR DESCRIPTION
**Pull request by ckeshava**
## High Level Overview of Change
Fixes https://github.com/XRPLF/xrpl-py/issues/503

It's ideal to validate a seed before creating a `Wallet` object. This is helpful in generating readable error messages for the users.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### Did you update CHANGELOG.md?

- [ ] Yes
- [x] No, this change does not impact library users
Invalid seed input to `Wallet` will throw a different error. The current code throws a `ValueError`, this is hard to understand for users.

## Test Plan
I have added a unit test that validates the Exception thrown upon empty seed inputs into the `Wallet` class constructor.

### CODE REVIEW

**Comment by ckeshava** [comment]
i don't understand why the `seed` is an `Optional[str]` in `Wallet`'s constructor here: https://github.com/XRPLF/xrpl-py/blob/a38f61d3c88b4a9a0807392ef4b0cb30aa688ce5/xrpl/wallet/main.py#L45

In what kind of situations can we get away with a non-existent seed? I saw a unit test that does not specify a `seed` for the `Wallet`, but I couldn't find any other examples. https://github.com/XRPLF/xrpl-py/blob/a38f61d3c88b4a9a0807392ef4b0cb30aa688ce5/tests/unit/asyn/wallet/test_main.py#L81

**Comment by ckeshava** [comment]
I see. So a seed is used to generate a public/private pair of keys. But I can bypass that step if I already know the cryptographic key pair.

File: tests/unit/asyn/wallet/test_main.py [hunk 1]
```diff
@@ -139,6 +140,15 @@ def test_constructor(self):
...
         _test_wallet_values(self, wallet, "seed", "secp256k1")
 
+    def test_wallet_contructor_throws_on_empty_seed(self):
+        with self.assertRaises(XRPLAddressCodecException):
+            Wallet(
+                constants["regular_key_pair"]["secp256k1"]["public_key"],
+                constants["regular_key_pair"]["secp256k1"]["private_key"],
+                master_address=constants["regular_key_pair"]["master_address"],
+                seed="",
+            )
```
Comment (JST5000) [inline tests/unit/asyn/wallet/test_main.py:150]: This test case can be simplified I believe by removing the master_address piece and using a clearly invalid seed (as opposed to one which could be mistaken for an "empty" seed - aka not passing one in)

```suggestion
    def test_wallet_contructor_throws_with_invalid_seed(self):
        with self.assertRaises(XRPLAddressCodecException):
            Wallet(
                constants["regular_key_pair"]["secp256k1"]["public_key"],
                constants["regular_key_pair"]["secp256k1"]["private_key"],
                seed="abc",
            )
```

Comment (JST5000) [reply tests/unit/asyn/wallet/test_main.py:150]: Or add another test case for the empty string case as opposed to the "regular" error case

**Review by JST5000 — APPROVED** [review]
Looks good other than a simplification in the test case :)

**Comment by ckeshava** [comment]
@justinr1234 can you please review this PR at your convenience?

**Review by justinr1234 — APPROVED** [review]
